src/app/components/trade/SwapPanel.tsx

```diff
@@ -87,16 +87,18 @@ export default function SwapPanel() {
 
   const handleInputPercentSelect = React.useCallback(
     (percent: number) => {
-      maxInputAmount && onPercentSelection(Field.INPUT, percent, maxInputAmount.amount.times(percent / 100).toFixed());
+      onPercentSelection(Field.INPUT, percent);
+      maxInputAmount && onUserInput(Field.INPUT, maxInputAmount.amount.times(percent / 100).toFixed());
     },
-    [onPercentSelection, maxInputAmount],
+    [onPercentSelection, onUserInput, maxInputAmount],
   );
 
   const handleOutputPercentSelect = React.useCallback(
     (percent: number) => {
-      maxInputAmount && onPercentSelection(Field.INPUT, percent, maxInputAmount.amount.times(percent / 100).toFixed());
+      onPercentSelection(Field.OUTPUT, percent);
+      maxInputAmount && onUserInput(Field.INPUT, maxInputAmount.amount.times(percent / 100).toFixed());
     },
```

Comment:
@WeiLi512 As Parrot9 said: "In the For field, the amount to fill for each % should be calculated based on the wallet balance of the Swap asset" => It seems we need to separating setInputValue function from selectPercent function. Because we need to set percentSelected for OUTPUT field and targetValue for INPUT field. Please help me check it again. Thank you!